### ISSUE TITLE
GET and HEAD should share the same Route

### ISSUE THREAD

**Issue by sebsel** [k-next/http#27]
I know this repo is a work in progress, but this might be nice to fix while you guys still can.
There is discussion on Kirby 2 here: https://github.com/getkirby/kirby/issues/538

As pointed out by @Zegnat in that discussion, GET and HEAD should return the same thing. 
It might be better to steer HEAD request into the GET routes, instead of having them as a separate array of Routes, as you [have now](https://github.com/k-next/http/blob/884623a8e90cb00ef223e92c6e102e08c28118c9/src/Http/Router.php#L39).

Pointing this out because HEAD requests might sound exotic, but are actually used a lot for Webmention, Micropub, IndieAuth, etc. At this moment in Kirby 2, custom Routes are returning 404 on a HEAD request, because GET is the default.

**Comment by lukasbestle** [comment]
I agree that we should "magically" support HEAD requests in the router. They should just work and it should not be possible to manually define a `HEAD` route (after all it is *always* supposed to return the same headers as the respective `GET` route).

Pseudocode:

```php
function router($method, $uri) {
	// if this is a HEAD request, find the route by the GET method
	$route = findRoute(($method === 'HEAD')? 'GET' : $method, $uri);
	
	if($method === 'HEAD') {
		// run the route, but discard all output
		ob_start();
		runRoute($route);
		ob_end_clean();
	} else {
		runRoute($route);
	}
}
```
